Feature/17: NCP Cloud Insight 메트릭 조회 기능 구현

unionxic opened the pull request:
## 개요
NCP Cloud Insight API를 연동하여 서버 인스턴스의 실시간 메트릭을 수집하고, 임계치 기반 알림 기능을 구현함. AWS/GCP와 동일한 패턴으로 NCP 메트릭 조회를 지원함.

## 변경 사항
* NCP 메트릭 조회 서비스 구현
  - `NcpMetricService`: Cloud Insight 메트릭 조회 서비스
  - CPU, Network, Disk, FileSystem 메트릭 지원
  - AWS CloudWatch, GCP Monitoring과 동일한 패턴 적용
* 메트릭 조회 엔드포인트 추가
  - `GET /api/ncp/accounts/{accountId}/servers/instances/{instanceNo}/metrics`
  - `NcpServerMetricsResponse` DTO로 메트릭 데이터 반환
* 알림 서비스 구현
  - `NcpAlertService`: 임계치 기반 알림 기능
  - 메트릭 임계값 초과 시 알림 트리거

## 배포
* 실제 배포 시 Cloud Insight API credentials 설정 필요
* NCP 계정별 메트릭 수집 권한 확인 필요
* 알림 임계값 설정 테스트 필요

## 참고
* Cloud Insight API 공식 문서 기반 구현
* 멀티 클라우드 메트릭 수집 아키텍처 일관성 유지
* 향후 프론트엔드 메트릭 차트 컴포넌트와 연동 가능
* 실시간 모니터링 및 비용 최적화 추천에 활용 예정

unionxic committed:
feat: NCP Cloud Insight 메트릭 조회 기능 구현

- Cloud Insight API 연동으로 서버 인스턴스 메트릭 수집
- AWS/GCP와 동일한 패턴으로 메트릭 조회 엔드포인트 추가
- CPU, Network, Disk, FileSystem 메트릭 지원
- 메트릭 기반 알림 서비스 구현

주요 변경사항:
- NcpMetricService: Cloud Insight 메트릭 조회 서비스
- NcpServerMetricsResponse: 메트릭 응답 DTO
- GET /api/ncp/accounts/{accountId}/servers/instances/{instanceNo}/metrics 엔드포인트 추가
- NcpAlertService: 임계치 기반 알림 기능

unionxic committed:
Merge branch 'develop' into feature/17

충돌 해결: NcpAlertService - 메트릭 기반 알림 기능이 포함된 feature/17 버전 사용

moon-seo commented:
```
log.info("NCP API Response: {}", response.toString());  
log.error("NCP GET 호출 실패: url={}, status={}, body={}", fullUrl, e.getStatusCode(), e.getResponseBodyAsString());
log.error("NCP POST 호출 실패: url={}, status={}, body={}", fullUrl, e.getStatusCode(), e.getResponseBodyAsString());
```

이거 로그 지워주셔야 할 것 같아용 노출 위험 있는 듯 합니다

unionxic committed:
fix: NCP API 민감 정보 로깅 제거

- NcpApiClient: API 응답 전체 로그 출력을 응답 길이만 출력하도록 변경
- NcpServerService: API 응답 로그 제거
- 보안 강화를 위해 액세스 키, 시크릿, 계정 정보 등이 노출될 수 있는 로그 제거

unionxic commented:
> ```
> log.info("NCP API Response: {}", response.toString());  
> log.error("NCP GET 호출 실패: url={}, status={}, body={}", fullUrl, e.getStatusCode(), e.getResponseBodyAsString());
> log.error("NCP POST 호출 실패: url={}, status={}, body={}", fullUrl, e.getStatusCode(), e.getResponseBodyAsString());
> ```
> 
> 이거 로그 지워주셔야 할 것 같아용 노출 위험 있는 듯 합니다

수정 완료했습니다!! 좋은 피드백 감사합니다!!
